streamer/src/nonblocking/quic.rs

```diff
@@ -729,11 +800,20 @@ async fn handle_connection(
     );
     let stable_id = connection.stable_id();
     stats.total_connections.fetch_add(1, Ordering::Relaxed);
-    let max_streams_per_100ms =
-        max_streams_for_connection_in_100ms(peer_type, params.stake, params.total_stake);
+    let mut max_streams_per_100ms = max_streams_for_connection_in_100ms(
+        peer_type,
+        params.stake,
+        params.total_stake,
+        stream_load_ema.clone(),
+    );
     let mut last_throttling_instant = tokio::time::Instant::now();
     let mut streams_in_current_interval = 0;
+    let staked_stream = matches!(peer_type, ConnectionPeerType::Staked) && params.stake > 0;
```

Comment:
why do we need to check both of these things this deep in the call stack? seems like it should've been resolved much higher

Reply:
When we initially did the QUIC implementation, I remember that peers with `stake = 0` were still marked as `ConnectionPeerType::Staked`. 
The inverse is not true though. I can change it to this. Would that be prefered?
```
let staked_stream = params.stake > 0;
```

Reply:
let's leave it as is for now and fix `ConnectionPeerType::Staked` where `params.stake == 0` in a followup. either the enum is misnamed or we're patching up a bug all over the place